students/kegan/session03/mailroom.py

```diff
@@ -0,0 +1,301 @@
+#!/usr/bin/env python3
+
+"""
+Kathryn Egan
+
+This program manages donors and their donations for
+Miuvenile Care, a charity organizaiton for disadvantaged
+kittens.
+"""
+
+
+DONORS = {
+    'Benedict Cumberbatch': [200000.0],
+    'Elon Musk': [10000.0, 150000.0, 100000.0],
+    'Dad': [20.0, 5.0],
+    'Donald Trump': [2.81],
+    'Billy Neighbor': [.54, .01, .25]}
+
+COLUMN_WIDTHS = [0] * 4
+
+
+def main():
+    """ Module that drives main menu."""
+    options = {
+        '1': {
+            'prompt': 'Send a Thank You',
+            'module': print_thank_you},
+        '2': {
+            'prompt': 'Create a Report',
+            'module': print_report},
+        '3': {
+            'prompt': 'Write Thank Yous',
+            'module': write_thank_yous},
+        '4': {
+            'prompt': 'Quit',
+            'module': exit_program}}
+    while True:
+        print('Please choose from the following options:')
+        prompt = '\n'.join(['{} {}'.format(
+            option, options[option]['prompt']) for option in options])
+        answer = input(prompt + '\n>')
+        if answer in options:
+            options[answer]['module']()
+
+
+def exit_program():
+    """ Exits program."""
+    from sys import exit
+    print('Exiting...')
+    exit()
+
+
+def print_thank_you():
+    """ Prompts user for donor name and amount and prints thank
+    you note to the console. Donation amount must be numerical."""
+    while True:
+        donor = input(
+            'Who donated? (Enter LIST to see current list of donors)\n>')
+        if donor.upper() != 'LIST':
+            break
+        print()
+        print('Current donors:')
+        print('\n'.join(sorted(DONORS)))
+        print()
+    donor = ' '.join(donor.split()).title()
+    if donor not in DONORS:
+        DONORS[donor] = []
+    donation = get_donation()
+    DONORS[donor].append(donation)
+    thankyou = get_thank_you(donor, [donation])
+    print(horizontal_line(80))
+    print(thankyou)
+    print(horizontal_line(80))
+
+
+def get_donation():
+    """ Prompts user for and returns donation amount.
+    Returns:
+        float : amount donated"""
+    while True:
+        donation = input('How much was donated?\n>')
+        try:
+            donation = float(donation.strip('$'))
+            return donation
+        except ValueError:
+            print('{} is not a valid amount.'.format(donation))
+            print('Please try again.')
+
+
+def horizontal_line(length):
+    """ Returns a horizontal line of given length.
+    Args:
+        length (int) : length of horizontal line
+    Returns:
+        str : horizontal line
+    """
+    return '-' * length
```

Comment:
really simple -- is it worth a function? MAybe -- if you think you may want to make it more complex at some point.